.github/workflows/upload-coverage-report.yml

```diff
@@ -0,0 +1,53 @@
+name: Upload Coverage Report
+
+on:
+  workflow_run:
+    workflows:
+      - LLPC Docker CI
+    types:
+      - completed
+
+jobs:
+  retrieve-and-upload:
+    if: github.repository == 'GPUOpen-Drivers/llpc' && github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.conclusion == 'success'
+    name: "Configuration: ${{ matrix.config_tag }}"
+    runs-on: "ubuntu-20.04"
+    strategy:
+      fail-fast: false
+      matrix:
+        config_tag:  ["release_clang_coverage", "release_clang_shadercache_coverage_assertions"]
+    steps:
+      - name: Authenticate with Google Cloud
+        id: 'auth'
+        uses: 'google-github-actions/auth@v0'
+        with:
+          credentials_json: '${{ secrets.GCR_KEY }}'
+      - name: Setup Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0
+      - name: Download code coverage artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: check-amdllpc-docker.yml
+          run_id: ${{ github.event.workflow_run.id }}
+          name: cov_report_${{ matrix.config_tag }}
+          path: "coverage_report/"
+      - name: Download PR number artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: check-amdllpc-docker.yml
+          run_id: ${{ github.event.workflow_run.id }}
+          name: pr_num
+          path: "pr_num/"
```

Comment:
Do these have access to our gcloud login? If yes, could we first download the artifacts and authenticate with gcloud later?

Reply:
I will move these before gcloud authentication to avoid possible security issues.